apisix/discovery/nacos.lua

```diff
@@ -0,0 +1,316 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local local_conf         = require("apisix.core.config_local").local_conf()
+local http               = require("resty.http")
+local core               = require("apisix.core")
+local ipairs             = ipairs
+local tostring           = tostring
+local type               = type
+local math               = math
+local math_random        = math.random
+local error              = error
+local ngx                = ngx
+local ngx_re             = require("ngx.re")
+local ngx_timer_at       = ngx.timer.at
+local ngx_timer_every    = ngx.timer.every
+local string             = string
+local string_sub         = string.sub
+local str_byte           = string.byte
+local str_find           = core.string.find
+local str_format         = string.format
+local log                = core.log
+
+local default_weight
+local applications
```

Comment:
We can initialize `applications = {}` so that we can save many codes like:

```
        if not applications then
            applications = up_apps
        end
```

Reply:
If use applications = {},run test could fail cause maybe call _M.nodes method before finish init.